packages/theme/tools/fetch-figma-variables.ts

```diff
@@ -0,0 +1,345 @@
+          return acc.concat("spacings")
+        default:
+          return acc.concat(cur)
+      }
+    }, [] as string[])
+
+    token.compatPath = compatPath;
+
+    return token;
+  },
+});
+
+/**
+ * Removes the base colors (color palette) from the final output.
+ */
+StyleDictionary.registerFilter({
+  name: 'filter-palette',
+  filter: (token: TransformedToken) => {
+    return token.prefix !== "color_palette"
+  }
+});
+
+/**
+ * Contents of the main TypeScript file linking the themes
+ */
+const tsIndex = `
+import light from "./light"
+import dark from "./dark"
+
+const themes = {
+  light,
+  dark
+}
+  
+export default themes`;
+
+/**
+ * Outputs a string to a file. Used for generated
+ * linking files defined above.
+ */
+StyleDictionary.registerFormat({
+  name: 'index',
+  format: async ({ file, options }) => {
+    const header = await fileHeader({ file });
+    return (
+      header
+      + options.content
+    );
+  },
+});
+
+/**
+ * Generates a nested JSON object using the path of each token as key.
+ *
+ * ["Color", "Background", "Neutral", "0"] becomes
+ * {
+ *   "Color": {
+ *     "Background": {
+ *       "Neutral": {
+ *         "0": value
+ *       }
+ *     }
+ *   }
+ * }
+ *
+ * @param tokens Flat list of design tokens
+ * @returns Nested object based on the path of each token
+ */
+const expandToNestedObject = (tokens: TransformedToken[], pathKey = 'path') => {
+  const result: any = {};
+  tokens.forEach((token) => {
+    let current = result;
+    token[pathKey].forEach((element: string, index: number) => {
+      if (typeof current === 'string') {
+        console.warn(`Converted ContrastColor to old format for ${token[pathKey].join('.')}`)
+        return
+      }
+
+      if (index === token[pathKey].length - 1) {
+        current[element] = token.value;
+      }
+      else {
+        current[element] = current[element] || {};
+        current = current[element];
+      }
+    });
+  });
+  return result;
+};
+
+/**
+ * Generates the nested JSON object and unquotes its keys.
+ */
+StyleDictionary.registerFormat({
+  name: 'typescript/obj',
+  format: async ({ dictionary, file, options }) => (`${await fileHeader({ file })
+    }export default ${JSON.stringify(expandToNestedObject(dictionary.allTokens, options.useFigmaStructure ? 'path' : 'compatPath'), null, 2).replace(/"([^"]+)":/g, '$1:')
+    };\n`),
+});
+
+/**
+ * Configures where the file should be output in accordance with the organization.
+ *
+ * @param organization Name of the organization
+ * @returns Output folder
+ */
+const makeDestination = (organization: Organizations, themeOptions?: ThemeOptions): string => path.join(outDir, `${themeOptions?.useFigmaStructure ? 'themes-fs' : 'themes'}/${organization}-theme/`);
+
+const generateThemes = async () => {
+
+  /**
+   * Mocked response from Figma Variables Rest API
+   */
+  const response = await fetch('https://api.figma.com/v1/files/3rlcixpbhfBglNSctUkMys/variables/local', {
```

Comment:
I think this url should also come from an env variable. I have added this in `FIGMA_VARIABLES_URL` as a repository variable (note: not secret) which you can use.

Reply:
Done!